src/test/java/bio/terra/workspace/integration/common/auth/AuthService.java

```diff
@@ -0,0 +1,79 @@
+package bio.terra.workspace.integration.common.auth;
+
+import bio.terra.workspace.integration.common.configuration.TestConfiguration;
+import com.google.auth.oauth2.AccessToken;
+import com.google.auth.oauth2.GoogleCredentials;
+import java.io.*;
+import java.util.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthService {
+
+  private final TestConfiguration testConfig;
+  private final Map<String, String> userTokens = new HashMap<>();
+  private final List<String> domainWideDelegationAccessScopes =
+      Arrays.asList(
+          "openid",
+          "email",
+          "profile",
+          "https://www.googleapis.com/auth/devstorage.full_control",
+          "https://www.googleapis.com/auth/cloud-platform");
+
+  @Autowired
+  public AuthService(TestConfiguration testConfig) {
+    this.testConfig = testConfig;
+  }
+
+  public String getAuthToken(String userEmail) throws IOException, InterruptedException {
+    if (!userTokens.containsKey(userEmail)) {
+      String vaultPath = testConfig.getVaultPath();
+      userTokens.put(userEmail, getDomainWideDelegationAccessToken(userEmail, vaultPath));
+    }
+    return userTokens.get(userEmail);
+  }
```

Comment:
I like the concept of using a token cache, but especially knowing that tokens have expirations, I'd prefer to use a proven cache implementation. I suggest [Caffeine](https://github.com/ben-manes/caffeine), which has proven itself in Rawls; Caffeine also is supported by Spring Cache, if we want Spring integration. Caffeine has support for automatically expiring values from the cache after N time units of writing them (https://github.com/ben-manes/caffeine/wiki/Eviction#time-based), which would assist with token expirations.

Reply:
That's very reasonable. Having now learned that token are valid for 15 minutes, would it make sense to remove the current cache, have the single createWorkspace test fetch token as it currently does, then fully implement a full-blown cache as part of another task?

Reply:
sure, works for me - remove the cache entirely, and if/when we need one, use Caffeine.

Reply:
Done!